File: app/src/main/java/com/github/goldy1992/mp3player/service/library/search/managers/SearchDatabaseManagers.java
package com.github.goldy1992.mp3player.service.library.search.managers;

import com.github.goldy1992.mp3player.commons.MediaItemType;

import java.util.EnumMap;

import javax.inject.Inject;
import javax.inject.Singleton;

@Singleton
public class SearchDatabaseManagers {

    final EnumMap<MediaItemType, SearchDatabaseManager> dbManagerMap;

    /**
     * Constructor
     * @param songDatabaseManager SongDatabaseManager
     * @param folderDatabaseManager FolderDatabaseManager
     */
    @Inject
    public SearchDatabaseManagers(SongDatabaseManager songDatabaseManager,
                                  FolderDatabaseManager folderDatabaseManager) {
        this.dbManagerMap = new EnumMap<>(MediaItemType.class);
        this.dbManagerMap.put(MediaItemType.SONGS, songDatabaseManager);
        this.dbManagerMap.put(MediaItemType.FOLDERS, folderDatabaseManager);
    }

    /**
     * reindexes all of the search database tables.
     */
    public void reindexAll() {
        for (SearchDatabaseManager manager : dbManagerMap.values()) {
            manager.reindex();
        }
    }
}

File: service/src/main/java/com/github/goldy1992/mp3player/service/library/search/managers/SearchDatabaseManagers.kt
package com.github.goldy1992.mp3player.service.library.search.managers

import com.github.goldy1992.mp3player.commons.MediaItemType
import dagger.hilt.android.scopes.ServiceScoped
import java.util.*
import javax.inject.Inject

@ServiceScoped
class SearchDatabaseManagers @Inject constructor(songDatabaseManager: SongDatabaseManager,
                                                 folderDatabaseManager: FolderDatabaseManager) {
    val dbManagerMap: EnumMap<MediaItemType, SearchDatabaseManager<*>>
    /**
     * reindexes all of the search database tables.
     */
    suspend fun reindexAll() {
        for (manager in dbManagerMap.values) {
            manager.reindex()
        }
    }

    /**
     * Constructor
     * @param songDatabaseManager SongDatabaseManager
     * @param folderDatabaseManager FolderDatabaseManager
     */
    init {
        dbManagerMap = EnumMap(MediaItemType::class.java)
        dbManagerMap[MediaItemType.SONGS] = songDatabaseManager
        dbManagerMap[MediaItemType.FOLDERS] = folderDatabaseManager
    }
}